src/main/kotlin/no/nav/familie/tilbake/vilkårsvurdering/VilkårsvurderingService.kt

```diff
@@ -0,0 +1,75 @@
+                              val foreldelseService: ForeldelseService,
+                              val faktaFeilutbetalingService: FaktaFeilutbetalingService) {
+
+    fun hentVilkårsvurdering(behandlingId: UUID): VurdertVilkårsvurderingDto {
+        val faktaOmFeilutbetaling = faktaFeilutbetalingService.hentAktivFaktaOmFeilutbetaling(behandlingId)
+                                    ?: throw Feil(message = "Fakta om feilutbetaling finnes ikke for behandling=$behandlingId, " +
+                                                            "kan ikke hente vilkårsvurdering")
+        val kravgrunnlag431 = kravgrunnlagRepository.findByBehandlingIdAndAktivIsTrue(behandlingId)
+        val vilkårsvurdering = vilkårsvurderingRepository.findByBehandlingIdAndAktivIsTrue(behandlingId)
+        val perioder = mutableListOf<Periode>()
+        val foreldetPerioder = mutableListOf<Periode>()
+        val vurdertForeldelse = foreldelseService.hentAktivVurdertForeldelse(behandlingId)
+        if (vurdertForeldelse == null) {
+            // fakta perioder
+            faktaOmFeilutbetaling.perioder.forEach { perioder.add(it.periode) }
+        } else {
+            // Ikke foreldet perioder
+            vurdertForeldelse.foreldelsesperioder.filter { !it.erForeldet() }
+                    .forEach { perioder.add(it.periode) }
+            // foreldet perioder
+            vurdertForeldelse.foreldelsesperioder.filter { it.erForeldet() }
+                    .forEach { foreldetPerioder.add(it.periode) }
+        }
+        return VilkårsvurderingMapper.tilRespons(vilkårsvurdering = vilkårsvurdering,
+                                                 perioder = perioder,
+                                                 foreldetPerioder = foreldetPerioder,
+                                                 faktaFeilutbetaling = faktaOmFeilutbetaling,
+                                                 kravgrunnlag431 = kravgrunnlag431)
+    }
+
+    @Transactional
+    fun lagreVilkårsvurdering(behandlingId: UUID, behandlingsstegVilkårsvurderingDto: BehandlingsstegVilkårsvurderingDto) {
+        val behandling = behandlingRepository.findByIdOrThrow(behandlingId)
+        val fagsystem = fagsakRepository.findByIdOrThrow(behandling.fagsakId).fagsystem
+        val kravgrunnlag = kravgrunnlagRepository.findByBehandlingIdAndAktivIsTrue(behandlingId)
+        // Valider request
+        VilkårsvurderingValidator.validerVilkårsvurdering(vilkårsvurderingDto = behandlingsstegVilkårsvurderingDto,
+                                                          kravgrunnlag431 = kravgrunnlag)
+        // filter bort perioder som er foreldet
+        val ikkeForeldetPerioder = behandlingsstegVilkårsvurderingDto.vilkårsvurderingsperioder
+                .filter { !foreldelseService.erPeriodeForeldet(behandlingId, Periode(it.periode.fom, it.periode.tom)) }
+        deaktiverEksisterendeVilkårsvurdering(behandlingId)
```

Comment:
trenger vi egentlig deaktivere vilkårsvurdering ved hver lagring? Hvorfor ikke en update?

Reply:
Det kan være vanskelig å gjøre en oppdate siden vilkårsvurdering har masse felt. Det er også en pattern som vi følger for hvert steg. Det implementeres samme i fptilbake også.